src/checkers/flymake-collection-mypy.el

```diff
@@ -32,28 +32,64 @@
 (eval-when-compile
   (require 'flymake-collection-define))
 
+(defcustom flymake-collection-mypy-args
+  '("--show-column-numbers"
+    "--no-error-summary"
+    "--no-color-output"
+    "--show-absolute-path"
+    "--show-error-codes")
+  "Command line arguments always passed to `flymake-collection-mypy'."
+  :type 'list
+  :group 'flymake-collection)
+
+(defcustom flymake-collection-mypy-root-files
+  '("mypy.ini" "pyproject.toml" "setup.cfg")
+  "Files used to guess the `default-directory' for invoking mypy.
+See `flymake-collection-mypy--default-directory' for details."
+  :type 'list
+  :group 'flymake-collection)
+
+(defun flymake-collection-mypy--default-directory (buffer)
```

Comment:
Please add an option to enable or disable this functionality and use the project library to find the project root instead of locate-dominating-file.

Reply:
Made it a bit more general: it can now try any of those options in order. Default is to look for config files, then try the project root, then fall back to the buffer `default-directory`. You can also use a directory path.